### ISSUE TITLE
"Read failed" importing Session Buddy session

### ISSUE THREAD

**Issue by Gooberpatrol66** [sienori/Tab-Session-Manager#912]
## Short description

The import button returns "Read failed" when given a session buddy session file

## Steps to reproduce

1. Export session from Session Buddy
2. Import session to Tab Session Manager

## Platform information
* Platform (OS): Linux
* Version of browser: Nightly 92.0.1
* Version of Tab Session Manager: 6.11.1

**Comment by lid** [comment]
I see the same issue on Firefox 94.0.2 and Tab Session Manager 6.11.1.

**Comment by lid** [comment]
Looking at #861 and e3179ca3c1aecea625918314673bd19baaa7743b, I think I understand the issue. In Session Buddy's export dialog the user can choose to export `selected`, `all`, `previous`, or `saved` sessions. If the export includes the `type == current` session (i.e. the active session that has not been saved), the exported file will not be recognized by the Tab Session Manager importer. This is because exported `current` sessions do not include all the fields listed in: https://github.com/sienori/Tab-Session-Manager/blob/e3179ca3c1aecea625918314673bd19baaa7743b/src/options/components/ImportSessionsComponent.js#L93

An exported `current` session will only include: `type`, `generated`, and `windows`, so at a glance it looks like tweaking `correctKeys` and handling a missing `SBSession.created` should resolve this.

Here's a sample file with a `current` session that generates this error in Tab Session Manager:
```
{
   "sessions": [
      {
         "type": "current",
         "generated": "2021-11-29T19:39:26.346Z",
         "windows": [
            {
               "alwaysOnTop": false,
               "focused": true,
               "height": 1391,
               "id": 1542,
               "incognito": false,
               "left": 1690,
               "state": "normal",
               "tabs": [
                  {
                     "active": false,
                     "audible": false,
                     "autoDiscardable": true,
                     "discarded": false,
                     "favIconUrl": "https://www.google.com/images/icons/product/chrome_web_store-32.png",
                     "groupId": -1,
                     "height": 1303,
                     "highlighted": false,
                     "id": 1543,
                     "incognito": false,
                     "index": 0,
                     "mutedInfo": {
                        "muted": false
                     },
                     "pinned": false,
                     "selected": false,
                     "status": "complete",
                     "title": "Session Buddy - Chrome Web Store",
                     "url": "https://chrome.google.com/webstore/detail/session-buddy/edacconmaakjimmfgnblocblbcdcpbko/related?hl=en",
                     "width": 1260,
                     "windowId": 1542
                  }
               ],
               "top": 43,
               "type": "normal",
               "width": 1276
            }
         ]
      }
   ]
}
```

**Comment by Nezteb** [comment]
> at a glance it looks like tweaking `correctKeys` and handling a missing `SBSession.created` should resolve this.

@lid 
[![futurama-you-are-technically-correct](https://user-images.githubusercontent.com/3588798/191897415-62af03bb-0d16-4a27-83b5-ab6f6e70309e.gif)](https://github.com/sienori/Tab-Session-Manager/pull/1071)

**Comment by sienori** [comment]
This issue was resolved by #1071. Thanks!
